### PR TITLE
Citation style

### PR DESCRIPTION
**Pull request by xUser5000**
The feature is not complete yet, but I decided to PR the changes I've done till now and get a review while working on the rest of the implementation. This saves a lot of time in my opinion. Please, don't merge this PR until I finish the initial implementation.

## What has been done (till now)
- Integrate [Citeproc.js](https://github.com/Juris-M/citeproc-js) with a thin layer of abstraction: `CSLProcessor` Singleton Class
- Register a new field in the config screen that enables the user to enter the path of a `.csl` file for styling. By default (if the field is empty), the APA style is used.
- Insert an empty line between references.

## Screenshots
APA
APA format

IEEE
IEEE format

## What's next
- Handle errors related to the `.csl` file (like "file not found" or "invalid format")
- Escape HTML entities

### CODE REVIEW

**Comment by laurent22** [comment]
Actually let me know when it's finished as I think it's easier that way. Otherwise my comments will be obsolete by the time you finish, or I might suggest changes that you're already planning to do.

**Comment by xUser5000** [comment]
Done the following:
- Handle errors related to the .csl file (like "file not found" or "invalid format")
- Escape HTML entities

File: src/ui/settings.ts [hunk 1]
```diff
@@ -1,33 +1,41 @@
...
-		iconName: PLUGIN_ICON
-	});
+    // Register settings fields
+    await joplin.settings.registerSettings({
+        [SETTINGS_BIBTEX_FILE_PATH_ID]: {
+            value: "",
+            type: SettingItemType.String,
+            section: SETTINGS_SECTION_ID,
+            public: true,
+            label: "BibTeX File",
```
Comment (laurent22) [inline src/ui/settings.ts:30]: In the description, you should mention that multiple files can be specified, separated by ;

File: src/ui/bibliography-renderer/index.ts [hunk 2]
```diff
@@ -1,39 +1,64 @@
...
+    setProcessorStyle(processor);
+
+    /**
+     * Listen to changes applied to the CSL field
+     * On change, set the style of the CSL Processor
+     */
+    joplin.settings.onChange(async (event) => {
+        if (event.keys.includes(SETTINGS_CSL_FILE_PATH_ID)) {
+            await setProcessorStyle(processor);
+            await joplin.views.dialogs.showMessageBox(MESSAGE_RESTART_APP);
```
Comment (laurent22) [inline src/ui/bibliography-renderer/index.ts:47]: Does that work even from the config screen? Actually, you could simply add a description below the field to say that the app needs to be restarted.

**Comment by xUser5000** [comment]
Done

**Pull request merged**